### ISSUE TITLE
Editor stuck in "The following files are newer on disk" After saving in external editor.

### ISSUE THREAD

**Issue by Logical-sh** [godotengine/godot#66644]
### Godot version

4.0.beta-2 (Mono)

### System information

Arch Linux x64 - Official Mono Build

### Issue description

Sometimes when saving code in my external editor, (vscode) I get a dialog saying the code file changed, and in this version (beta 2) it just reopens the dialog infinitely until I chose to resave the file. This did not happen in beta 1, I was able to just click reload.
Now when I try to click reload I get this in the debug console and the dialog just reopens.
`core/object/object.cpp:1057 - Error calling from signal 'confirmed' to callable: 'ScriptEditor::ScriptEditor::reload_scripts': Method expected 1 arguments, but called with 0.`

### Steps to reproduce

Edit code file in external editor and save, click back into godot to run.

**Comment by akien-mga** [comment]
CC @Rindbee.

**Comment by Rindbee** [comment]
A temporary solution is to enable `text_editor/behavior/files/auto_reload_scripts_on_external_change`.

**Comment by Calinou** [comment]
> A temporary solution is to enable `text_editor/behavior/files/auto_reload_scripts_on_external_change`.

Out of curiosity, why is this disabled by default?

**Comment by rhofour** [comment]
@Rindbee I don't seem to have that setting at all. In a fresh project started in 4.0 (Beta 2) I don't have the whole Text Editor section of the settings. When I try and add a new boolean setting with that name it doesn't seem to fix this issue.

Should I report the missing setting as a separate bug?

**Comment by Rindbee** [comment]
> I don't seem to have that setting at all. In a fresh project started in 4.0.beta2 I don't have the whole Text Editor section of the settings. When I try and add a new boolean setting with that name it doesn't seem to fix this issue.

If you don't mind resetting the entire editor settings, you can go through **Editor** - **Open Editor Settings Folder,** then delete the **editor_settings-4.tres** file. Restart the editor and reset this option.

**Comment by rhofour** [comment]
Oh wait, this was my mistake! I was looking in project settings and not editor settings. I do in fact have the right setting and it does work around this issue. Sorry about that!

**Comment by tsarquis88** [comment]
Same issue here! 
The error happens even when I write a config file with `ConfigFile.save(path)`.

4.0.beta-2 version on Debian bullseye

**Comment by Skwint** [comment]
This is happening to me and I can't even alt-f4 away from godot. In my case the root cause is renaming a tscn file in a way that changed only the case of one letter, and renaming the .gd file to go with it. The file it is saying is newer on disk is the .gd file and has the old case. The editor shows both new case and old case in the list of files I have open.

**Comment by Rindbee** [comment]
In this case, any of the three buttons will form an infinite popup loop (see #65824), which should be fixed by #66812.

**Comment by viktor-ferenczi** [comment]
We had this problem with the shaders editors, which has been fixed recently. Maybe a similar issue.

**Comment by Retorikal** [comment]
> A temporary solution is to enable `text_editor/behavior/files/auto_reload_scripts_on_external_change`.

On 4.0.beta3.mono, using this workaround makes an issue identical to #49298 despite the issue being closed. The edits made on vscode aren't recognized, only after disabling _General/Text Editor/External/Use External Editor_ and reopening it in internal editor the changes are recognized.

Attached is a project that exhibits the problem. Reproducing steps are identical with the mentioned issue and the attempted changes are commenting out the exported variables.
[Troubleshoot.zip](https://github.com/godotengine/godot/files/9832416/Troubleshoot.zip)

**Comment by Rindbee** [comment]
> On 4.0.beta3.mono (Ubuntu 22.04), using this workaround makes an issue identical to #49298 despite the issue being closed. The edits made on vscode aren't recognized, only after disabling _General/Text Editor/External/Use External Editor_ and reopening it in internal editor the changes are recognized.

I modified the relevant code in #66867 and partially reverted #66405. So it's back to what it was before.

I try to use a signal to update the internal editor. Implemented using a series of PRs, see https://github.com/godotengine/godot/pull/66658#issue-1392290784, but at present it's only partially applied (#66867 is also part of this series). So, the effect is not as expected. 

`ScriptEditor` just uses the script list to detect file changes. It is not as good as `EditorFileSystem` in detecting files. So my plan is to use `EditorFileSystem` to detect script file changes, and then pass the list of changed script files to some class (`ScriptEditServer`? Makes it possible to disable `ScriptEditor`) to process. #66880 should make this a little easier.
